selenium/codenvy-selenium-test/src/test/java/com/codenvy/selenium/factory/AuthenticateAndAcceptFactoryThroughGitHubOAuthTest.java

```diff
@@ -56,38 +61,37 @@
   @Inject private TestApiEndpointUrlProvider apiEndpointUrlProvider;
   @Inject private TestUserNamespaceResolver testUserNamespaceResolver;
   @Inject private TestWorkspaceServiceClientFactory testWorkspaceServiceClientFactory;
+  @Inject private DashboardWorkspace dashboardWorkspace;
+  @Inject private TestGitHubServiceClient testGitHubServiceClient;
 
   private TestFactory testFactory;
+  private final String WORKSPACE_TOOLBAR = "Workspaces";
 
   @BeforeClass
   public void setUp() throws Exception {
     testFactory = testFactoryInitializer.fromTemplate(FactoryTemplate.MINIMAL).build();
   }
 
-  //  @AfterClass
-  // This method removes default user instead of github user.
-  // Need to be reworked https://github.com/codenvy/codenvy/issues/2471
+  @AfterClass
   public void tearDown() throws Exception {
-    User user = testUserServiceClient.findByEmail(testUser.getEmail());
-    TestWorkspaceServiceClient workspaceServiceClient =
-        testWorkspaceServiceClientFactory.create(testUser.getEmail(), testUser.getPassword());
-    workspaceServiceClient
-        .getAll()
-        .forEach(
-            ws -> {
-              try {
-                workspaceServiceClient.delete(ws, user.getName());
-              } catch (Exception e) {
-                throw new RuntimeException(e);
-              }
-            });
 
-    testUserServiceClient.remove(testUserServiceClient.findByEmail(testUser.getEmail()).getId());
+    seleniumWebDriver.get(getDashboardWorkspaceUrl());
```

Comment:
It could be simplified by using `dashboard.selectWorkspacesItemOnDashboard();` command.

Reply:
Yes, I tried to do as you wrote, but it not worked well in "@AfterClass" method , so I used another construction.

Reply:
Movement of code into the **DoashboardWorkspaces** page object will get rid of **getDashboardWorkspaceUrl()** method.